Formula/gdal.rb

```diff
@@ -138,12 +139,9 @@ def install
       "--without-sosi",
     ]
 
-    # Work around "error: no member named 'signbit' in the global namespace"
-    # Remove once support for macOS 10.12 Sierra is dropped
-    if DevelopmentTools.clang_build_version >= 900
-      ENV.delete "SDKROOT"
-      ENV.delete "HOMEBREW_SDKROOT"
-    end
+    # Use macOS SDK for headers for 10.14 and above (SDK version must match OS version)
+    ENV["HOMEBREW_SDKROOT"] = MacOS::CLT.sdk_path(MacOS.version) if MacOS.version >= :mojave && MacOS::CLT.installed?
+    ENV["SDKROOT"] = MacOS::CLT.sdk_path(MacOS.version) if MacOS.version >= :mojave && MacOS::CLT.installed?
```

Comment:
This will give you a versioned SDK path on Big Sur, which we've been trying not to do: https://github.com/Homebrew/brew/pull/10072, https://github.com/Homebrew/homebrew-core/issues/67075

I'm pretty sure `gdal` bakes in the SDK path too, so these two lines are going to be a bit of a headache for the next Big Sur minor version bump. https://github.com/Homebrew/homebrew-core/issues/68049